core/tracepoint/binding_spec.rb

```diff
@@ -0,0 +1,11 @@
+require File.expand_path('../../../spec_helper', __FILE__)
+
+ruby_version_is '2.4' do
+  describe 'TracePoint#binding' do
+    it 'return the generated binding object from event' do
+      binding = nil
+      TracePoint.new(:line) { |tp| binding = tp.binding }.enable
+      binding.should be_kind_of(Binding)
```

Comment:
This spec needs to be more specific as to which binding is being returned. If I'm not mistaken, we could compare the binding return with the current binding, no?

Reply:
I am new to this. I didn't get this. Can you please explain?

Reply:
It can be confusing to work with `binding`, but they can be powerful. For example, one can access a local variable, so with this test one could write:
```
def foo
  local = 42
  return :hello
end
```
If I'm not mistaken, if you trace `:return`, the binding should give you access to `local`

Reply:
I tried & found that if we trace `:return` the binding doesn't give us access to local -- 
```
binding = nil

def test   
  local = 4
  return :hello
end

TracePoint.new(:return) do |tp|
  binding = tp.binding
end.enable

test
p binding.local_variables => [:matcher, :expected, :actual]
```
If binding was available on the method then local_variables would have had `local` included.

Reply:
Mmm, I think that `mspec` might be playing dirty tricks on you. If you run the code you wrote in Ruby, it works as expected. I'm too tired right now to delve into this deeper though. Good night

Reply:
Maybe one way to avoid the problem here is is to enable just around the call:
```ruby
TracePoint.new(:return) do |tp|
  binding = tp.binding
end.enable { test }
```
If that's not enough, then assigning binding if it's not already set should help (`binding = tp.binding unless binding`).
@Atul9 Could you try that?

Reply:
I pushed af2c4217da2c7493f59a28dd79022662d685fbcd to fix this.